### PR TITLE
DiscoveryReport templates

### PR DESCRIPTION
**Pull request by atz**
These are stub templates, but show structure and test it.  I don't mind if reviewers prefer to wait for them to be further fleshed out, but they are mergeable as is.  Since there is some confusion about how these components might work together, this may be clarifying.

Note: because of the Enumerator, we can use the form:
```ruby
render partial: 'row', collection: @discovery_report.each_row
```
That knows to iterate over the Enumerator.

We can also imagine other template outputs (JSON, HTML, etc.) based on the same structure.

### CODE REVIEW

File: app/services/discovery_report.rb [hunk 1]
```diff
@@ -70,26 +66,19 @@ def registration_check(druid)
...
-    fields << 'SourceID unique in DOR?' if using_manifest?
+    fields.concat ['Duplicate Filenames?', 'DRUID', 'Registered?', 'APO exists?', 'SourceID unique in DOR?']
     fields.join(' , ')
   end
 
   # For use by template
   def skipped_files
     files = ['Thumbs.db', '.DS_Store'] # if these files are in the bundle directory but not in the manifest, they will be ignorned and not reported as missing
     files << File.basename(content_md_creation[:smpl_manifest]) if using_smpl_manifest?
-    files << File.basename(manifest) if using_manifest?
```
Comment (ndushay) [inline app/services/discovery_report.rb]: I thought we were requiring a manifest file in all cases -- so the if clause can go, but the line should stay?

Comment (ndushay) [reply app/services/discovery_report.rb]: ^^ whoops - I guess the above is a "request changes"

Comment (atz) [reply app/services/discovery_report.rb]: Thanks, incorporated 2 lines earlier in the initial array.

**Comment by atz** [comment]
Yes, I think your projection of the ultimate goal is correct.  But to be clear, this PR is not going to get into:
1. writing an output file
2. testing an output file
3. anything involving jobs
4. including sending an email

So there is no payoff to holding out for those things here.  The next step (that *could* be in PR or subsequent) would be making the partial template for `row` dynamic (i.e. not just an example line).

File: app/services/discovery_report.rb [hunk 2]
```diff
@@ -70,26 +66,19 @@ def registration_check(druid)
   # @return [String] primitive version
   def header
     fields = ['Object Container', 'Number of Items', 'Files with no ext', 'Files with 0 Size', 'Total Size', 'Files Readable']
-    fields.concat ['Label', 'Source ID'] if using_manifest?
+    fields.concat ['Label', 'Source ID']
     fields.concat ['Num Files in CM Manifest', 'All CM files found'] if using_smpl_manifest?
-    fields.concat ['Duplicate Filenames?', 'DRUID', 'Registered?', 'APO exists?']
-    fields << 'SourceID unique in DOR?' if using_manifest?
+    fields.concat ['Duplicate Filenames?', 'DRUID', 'Registered?', 'APO exists?', 'SourceID unique in DOR?']
```
Comment (ndushay) [inline app/services/discovery_report.rb]: whoops - can get rid of "SourceID unique in DOR?" header -- sorry I didn't notice on first pass.

File: app/services/discovery_report.rb [hunk 3]
```diff
@@ -70,26 +66,19 @@ def registration_check(druid)
...
     fields.join(' , ')
   end
 
   # For use by template
   def skipped_files
-    files = ['Thumbs.db', '.DS_Store'] # if these files are in the bundle directory but not in the manifest, they will be ignorned and not reported as missing
+    files = ['Thumbs.db', '.DS_Store', manifest] # if these files are in the bundle directory but not in the manifest, they will be ignorned and not reported as missing
```
Comment (ndushay) [inline app/services/discovery_report.rb]: looks good.  slight preference for manifest first (and I think :manifest might need to get a delegation in bundle.db?  Also the comment reads a little weird now (and has a typo) -- maybe just change it to a method comment?

File: app/services/discovery_report.rb (hunk 2 above)
Comment (atz) [reply app/services/discovery_report.rb]: Seems like we can get rid of headers completely, if we are making the report output JSON.

**Comment by atz** [comment]
I'm cool w/ merging this as is, since from our conversations, I gather that what we are really after is more structured JSON output (not plaintext).

**Comment by ndushay** [comment]
merging.  Assume you'll address the need to delegate :manifest in bundle.rb in another PR.

**Pull request merged**